[DOC] Added study spec (i.e. dir layout) to the explanations page (#815)
nipoppy/nipoppy: docs/source/explanations/study_spec.md

# Study specification

> Nipoppy provides a specification (i.e. directory layout) for dataset organization that extends the [Brain Imaging Data Structure (BIDS) standard](https://bids.neuroimaging.io/). One can think of this as a "standard" for organizing data generated during a lifecycle of a `study` which may include `sourcedata` (e.g. DICOMs), `tabular` (e.g., phenotypic assessments), and imaging `derivatives` (e.g. fmriprep output), in addition to the raw BIDS data. A `study`-level standardized directory layout simplifies curation and processing tasks as well as tracking of outputs. Additionally, the layout consists of directories to store pipeline specifications and custom code used for data processing and wrangling tasks. The complete layout configuration details are provided [here](../schemas/index.rst).

Reviewer: > The complete layout configuration details are provided [here](../schemas/index.rst).

This is more for configuring a custom layout. Do we want this to be part of this docs page?

Also, the link points to the whole page but we can make it point to the specific subheading, see how it's done [here](https://github.com/nipoppy/nipoppy/blob/10876d6246818212cb2e3ab15745869d68319858/docs/source/how_to_guides/user_guide/global_config.md?plain=1#L86).

Author: Okay will remove it.

> There are on-going efforts to merge this specification as a `study`-level dataset type into the BIDS standard.

Reviewer: Rewording suggestion:

```suggestion
There are ongoing efforts to merge this specification with the BIDS standard's study-level dataset type.
```

> The directory tree shown below is generated automatically upon dataset initialization (see [nipoppy init](../how_to_guides/init/index.md)).

Reviewer: Linking to the CLI reference page makes more sense, no?

```suggestion
The directory tree shown below is generated automatically upon dataset initialization (see [`nipoppy init`](../cli_reference/init.rst)).
```
